doc/API.md

```diff
@@ -619,97 +623,106 @@ struct {
 
 Parameters:
 ```
-path        string
-destination string
+siapaths []string
 ```
-'path' is the nickname of the file.
-
-'destination' is the path that the file will be downloaded to.
+'siapaths' is an array of the nicknames to be shared. It is comma-delimited.
 
-Response: standard
+Response:
+```
+struct {
+	asciisia string
+}
+```
+'asciisia' is the ASCII-encoded .sia file.
 
-#### /renter/rename/{path} [POST]
+#### /renter/delete/{siapath} [POST]
 
-Function: Rename a file. Does not rename any downloads or source files, only
-renames the entry in the renter.
+Function: Deletes a renter file entry. Does not delete any downloads or
+original files, only the entry in the renter.
 
 Parameters:
 ```
-nickname string
-newname  string
+siapath string
 ```
-'nickname' is the current name of the file entry.
-
-'newname' is the new name for the file entry.
+'siapath' is the location of the file in the renter.
 
-Response: standard.
+Response: standard
 
-#### /renter/share [GET]
+#### /renter/download/{siapath} [GET]
 
-Function: Create a .sia file that can be shared with other people.
+Function: Downloads a file. The call will block until the download completes.
 
 Parameters:
 ```
-paths       []string
+siapath     string
 destination string
 ```
-'paths' is an array of the nicknames to be shared. It is comma-delimited.
+'siapath' is the location of the file in the renter.
 
-'destination' is the path of the .sia file to be created. It must end in
-'.sia'.
+'destination' is the location on disk that the file will be downloaded to.
 
-Response: standard.
+Response: standard
 
-#### /renter/shareascii [GET]
+#### /renter/rename/{siapath} [POST]
 
-Function: Create an ASCII .sia file that can be shared with other people.
+Function: Rename a file. Does not rename any downloads or source files, only
+renames the entry in the renter.
 
 Parameters:
 ```
-paths []string
+siapath     string
+newsiapath  string
 ```
-'paths' is an array of the nicknames to be shared. It is comma-delimited.
+'siapath' is the current location of the file in the renter.
 
-Response:
-```
-struct {
-	file string
-}
-```
-'file' is the ASCII-encoded .sia file.
+'newsiapath' is the new location of the file in the renter.
 
-#### /renter/upload/{path} [POST]
+Response: standard.
+
+#### /renter/upload/{siapath} [POST]
 
 Function: Uploads a file.
 
 Parameters:
 ```
-path     string
+siapath  string
 source   string
+duration types.BlockHeight (uint64)
+renew    bool
 ```
-'path' is the nickname that will be used to reference the file.
+'siapath' is the location where the file will reside in the renter.
+
+'source' is the location on disk of the file being uploaded.
+
+'duration' is the number of blocks for which the file will be available. If
+the renew parameter is true, this parameter will be ignored.
 
-'source' is the path to the file to be uploaded.
+'renew' indicates whether the file's contracts should be automatically renewed
```

Comment:
How does this get cancelled? Should the UI provide both options and if so have a default duration or renew by default?